File: api/internal/domain/notification/notification.go
package notification

import "time"

type Channel string

const (
	ChannelInApp    Channel = "in_app"
	ChannelEmail    Channel = "email"
	ChannelWhatsapp Channel = "whatsapp"
)

type Notification struct {
	ID        string
	UserID    string
	Type      string
	Title     string
	Body      string
	Data      map[string]any
	ReadAt    *time.Time
	CreatedAt time.Time
}

File: admin/src/types/notification.ts
import type { PaginatedResponse } from "@/types/api";

export interface Notification {
  id: string;
  type: string;
  title: string;
  body: string;
  data?: Record<string, unknown>;
  readAt: string | null;
  createdAt: string;
}

export interface NotificationUnreadCountResponse {
  count: number;
}

export type PaginatedNotificationsResponse = PaginatedResponse<Notification>;
